cirq/ops/measurement_gate.py

```diff
@@ -64,7 +64,10 @@ def __init__(
         self._qid_shape = qid_shape
         if len(self._qid_shape) != num_qubits:
             raise ValueError('len(qid_shape) != num_qubits')
-        self.key = key
+        if isinstance(key, str):
+            self.key = key
+        else:
+            raise TypeError('Wrong type for measurement key. MeasurementGate only accepts str key')
```

Comment:
Pretty sure we'd need to allow tuples here for when you flatten a circuit that has tuples produced by subcircuits. The equivalent flattened circuit would have to measure directly into the tuple.